Managers/PrimitiveMan.h

```diff
@@ -0,0 +1,546 @@
+			short m_Alignment; //! Alignment of text.
+			std::string m_Text; //! String containing text to draw.
+
+			/// <summary>
+			/// Constructor method for TextPrimitive object.
+			/// </summary>
+			TextPrimitive() {
+				m_IsSmall = true;
+				m_Text = "";
+				m_Alignment = 0;
+				m_Player = -1;
+			}
+
+			/// <summary>
+			/// Constructor method for TextPrimitive object.
+			/// </summary>
+			TextPrimitive(Vector pos, std::string text, bool isSmall, short alignment) {
+				m_StartPos = pos;
+				m_Text = text;
+				m_IsSmall = isSmall;
+				m_Alignment = alignment;
+				m_Player = -1;
+			}
+
+			/// <summary>
+			/// Constructor method for TextPrimitive object.
+			/// </summary>
+			TextPrimitive(short player, Vector pos, std::string text, bool isSmall, short alignment) {
+				m_StartPos = pos;
+				m_Text = text;
+				m_IsSmall = isSmall;
+				m_Alignment = alignment;
+				m_Player = player;
+			}
+
+			/// <summary>
+			/// Draws this primitive on provided bitmap.
+			/// </summary>
+			/// <param name="pDrawScreen">Bitmap to draw on.</param>
+			/// <param name="targetPos">Position of graphical primitive.</param>
+			virtual void Draw(BITMAP *pDrawScreen, Vector targetPos);
+		};
+#pragma endregion
+
+#pragma region Bitmap Primitive
+		/// <summary>
+		/// Class used to schedule drawing of bitmap primitives created from Lua.
+		/// </summary>
+		class BitmapPrimitive : public GraphicalPrimitive {
+
+		public:
+
+			float m_RotAngle; //! Angle to rotate bitmap in radians.
+			BITMAP *m_pBitmap; //! Bitmap to draw.
+
+			/// <summary>
+			/// Constructor method for BitmapPrimitive object.
+			/// </summary>
+			BitmapPrimitive() {
+				m_pBitmap = 0;
+				m_RotAngle = 0.0;
+				m_Player = -1;
+			}
+
+			/// <summary>
+			/// Constructor method for BitmapPrimitive object.
+			/// </summary>
+			BitmapPrimitive(Vector pos, BITMAP * bitmap, float rotAngle) {
+				m_StartPos = pos;
+				m_pBitmap = bitmap;
+				m_RotAngle = rotAngle;
+				m_Player = -1;
+			}
+
+			/// <summary>
+			/// Constructor method for BitmapPrimitive object.
+			/// </summary>
+			BitmapPrimitive(short player, Vector pos, BITMAP * bitmap, float rotAngle) {
+				m_StartPos = pos;
+				m_pBitmap = bitmap;
+				m_RotAngle = rotAngle;
+				m_Player = player;
+			}
+
+			/// <summary>
+			/// Draws this primitive on provided bitmap.
+			/// </summary>
+			/// <param name="pDrawScreen">Bitmap to draw on.</param>
+			/// <param name="targetPos">Position of graphical primitive.</param>
+			virtual void Draw(BITMAP *pDrawScreen, Vector targetPos);
+		};
+#pragma endregion
+#pragma endregion
+
+#pragma region Creation
+		/// <summary>
+		/// Constructor method used to instantiate a PrimitiveMan object in system memory.
+		/// </summary>
+		PrimitiveMan() { ClearPrimitivesList(); }
+#pragma endregion
+
+#pragma region Destruction
+		/// <summary>
+		/// Delete all scheduled primitives, called on every FrameMan sim update.
+		/// </summary>
+		void ClearPrimitivesList();
```

Comment:
This should maybe have an explicit destructor that calls ClearPrimitivesList(). Not a big deal for the same reasons as PostProcessMan, since the built-in destructor definitely does everything you need anyway.